subsys/bluetooth/audio/ascs.c

```diff
@@ -1378,6 +1378,68 @@ static int ase_config(struct bt_ascs *ascs, struct bt_ascs_ase *ase,
 	return 0;
 }
 
+int bt_audio_unicast_server_config_ase(struct bt_conn *conn,
+		struct bt_audio_stream *stream, struct bt_codec *codec,
+		const struct bt_codec_qos_pref *qos_pref)
+{
+	struct bt_ascs *ascs;
+	struct bt_ascs_ase *ase;
+	struct bt_audio_ep *ep;
+
+	CHECKIF(conn == NULL || stream == NULL || codec == NULL ||
+		qos_pref == NULL) {
+		LOG_DBG("NULL value(s) supplied)");
+		return -EINVAL;
+	}
+
+	ascs = ascs_get(conn);
+	ep = stream->ep;
+
+	if (ep == NULL) {
+		sys_snode_t *ase_node;
+		uint8_t ase_next_id = 1;
+
+		SYS_SLIST_FOR_EACH_NODE(&ascs->ases, ase_node) {
+			ase_next_id++;
+		}
+
+		ase = ase_get(ascs, ase_next_id);
+		if (ase == NULL) {
+			LOG_WRN("No free ASE found for config ASE ID 0x%02x", ase_next_id);
+			return -ENOTSUP;
+		}
+
+		ep = &ase->ep;
+	}
+
+	switch (ep->status.state) {
+	/* Valid only if ASE_State field = 0x00 (Idle) */
+	case BT_AUDIO_EP_STATE_IDLE:
+	 /* or 0x01 (Codec Configured) */
+	case BT_AUDIO_EP_STATE_CODEC_CONFIGURED:
+	 /* or 0x02 (QoS Configured) */
+	case BT_AUDIO_EP_STATE_QOS_CONFIGURED:
+		break;
+	default:
+		LOG_ERR("Invalid state: %s", bt_audio_ep_state_str(
+			ep->status.state));
+		return -EBADMSG;
+	}
```

Comment:
`bt_audio_unicast_server_config_ase` shall configure a new ASE. If the ASE is any other state than idle, then we should return an error.